src/model/chunk.py

```diff
@@ -0,0 +1,36 @@
+## This file contains the function to chunk the input text into smaller pieces for the model to process
+## The input text is tokenized and then split into chunks of size chunk_size
+
+from transformers import AutoTokenizer
+import torch
+
+def chunk(inp: str, 
+          tokenizer: AutoTokenizer, 
+          chunk_size: int = 256, 
+          overlapping_len: int = 3,
+          num_chunks: int = 128) -> list:
+    tokenized_txt = tokenizer(inp, return_tensors="pt")["input_ids"]
+    token_len = tokenized_txt.size()[1]
+    
+    if token_len <= chunk_size:
+        return [tokenized_txt]
+    else:
+        chunks = []
+        chunks.append(tokenized_txt.view(-1)[:chunk_size])
+        for i in range(chunk_size, token_len, chunk_size):
+            if len(chunks) == num_chunks:
+                # only return num_chunks chunks so that it is not stuck on the same sample for too long
+                break
+            if i + chunk_size > token_len:
+                new_chunk = tokenized_txt.view(-1)[i-overlapping_len:]
+                new_chunk_len = new_chunk.size()[0]
+                new_chunk = new_chunk.view(1, new_chunk_len)
```

Comment:
not the end of the world but the fact that there is a view(-1) and a view(1) means at least one of these is an unnecessary operation. this can be made more efficient -- not a big deal for this project but could end up being a significant overhead with a large enough `chunk_size`

Reply:
Resolved in https://github.com/MichiganDataScienceTeam/F24-mini-copilot/pull/24/commits/926beedac4e3a431f36582d81dae250ddf5e172f